### DISCUSSION TITLE
Docker container suddenly stopped working - ring-mqtt Network is offline, waiting 10 seconds to check again... +15s

### DISCUSSION THREAD

**Discussion by Mario084** [tsightler/ring-mqtt#808] [Support Q&A]
Hello folks!
My ring mqtt integration stopped working after a year now. I realized that there is no status update in openhab anymore. I checked the docker container and saw that it has status "stopped". 

When I try to restart I get the following entries in the terminal:

`
| Ring-MQTT with Video Streaming                      |
| Docker Edition                                      |
|                                                     |
| For support questions please visit:                 |
| https://github.com/tsightler/ring-mqtt/discussions  |
-------------------------------------------------------
cont-init: info: /etc/cont-init.d/ring-mqtt.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun ring-mqtt (no readiness notification)
s6-rc: info: service legacy-services successfully started
-------------------------------------------------------
ring-mqtt.js version: 5.0.5
Node version v16.17.1
NPM version 8.10.0
git version 2.36.2
-------------------------------------------------------
Running ring-mqtt...
  ring-mqtt Detected runmode: docker +0ms
  ring-mqtt Configuration file: /data/config.json +6ms
  ring-mqtt MQTT URL: mqtt://openhabian:********@192.168.1.199:1883 +6ms
  ring-mqtt Reading latest data from state file: /data/ring-state.json +0ms
  ring-mqtt Network is offline, waiting 10 seconds to check again... +0ms
  ring-mqtt Network is offline, waiting 10 seconds to check again... +15s
  ring-mqtt Network is offline, waiting 10 seconds to check again... +15s
  ring-mqtt Network is offline, waiting 10 seconds to check again... +15s

`

Any idea how to fix this? I´m not sure if it´s a problem with the docker container or my network? 
Camera is of course up and running. Using latest docker container.

Many thanks in advance!

Br, Mario

**Comment by tsightler** [comment]
Not being able to reach the network is an issue with the network, not ring-mqtt or the container.  Could be caused by anything that impacts the network.  The project uses the is-online package to determine if the public Internet can be reached prior to attempting connection to Ring.  This is done for cases where there might be a network outage (think a power outage or something) and the HA system boots before the network connection is restored.

**Reply by CarnivoreD** [reply]
Hi

Unfortunately I have the same problem. I have a Ring device that I have connected via Ring MQTT as a Docker container. Everything worked fine until I suddenly stopped receiving MQTT messages.

When I (re)start the container, I get the following message :

WARNING: IPv4 forwarding is disabled. Networking will not work.

The following message appears in the container terminal every 10 seconds:

2024-06-12T14:39:08.529Z ring-mqtt Network is offline, waiting 10 seconds to check again...

BUT: The network settings correspond to those of other containers that run without problems.

Even a complete new setup didn't help. How can I get it back working?

Many thanks and best...

**Reply by tsightler** [reply]
The message is really clear, IPv4 forwarding is disabled.  That is a message from Docker, not a message from ring-mqtt, but ring-mqtt does require the ability to be able to reach the network and, if Docker detects that IPv4 forwarding is disabled, the container will not be able to reach the network.  You should research how to solve this Docker host configuration issue as it has nothing to do with ring-mqtt.  Most likely you are using a docker network but net.ipv4.ip_forward is not set to 1, but there can be many other reasons, however, ring-mqtt will not be able to function until you solve that problem.

**Reply by CarnivoreD** [reply]
Thank you very much - it is helpful to know, that IPv4 forwarding is required...

What helped is to stop the docker-service and restart it.
